### PR TITLE
16 implement pathlib

### PR DESCRIPTION
**Pull request by PipGrylls**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please add a line in the relevant section of [CHANGELOG.md](https://github.com/paramm-team/pybamm-param/blob/main/CHANGELOG.md) to document the change (include PR #) - note reverse order of PR #s. If necessary, also add to the list of breaking changes.

- [ ] New feature (non-breaking change which adds functionality)
- [x] Optimization (back-end change that speeds up the code)
- [ ] Bug fix (non-breaking change which fixes an issue)

## Key checklist

- [x] No style issues: `$ python -m nox -s lint`
- [] All tests pass: `$ python -m nox -s unit examples integration`
- [x] Coverage is complete `$ python -m nox -s coverage`
- [x] The documentation builds: `$ python -m nox -s sphinx`

## Further checks

- [x] Code is commented, particularly in hard-to-understand areas
- [~] Tests added that prove fix is effective or that feature works

@brosaplanella 
- Can you look over the cycler keywords and make sure that you are happy with them. The ‘TestTime’ keyword in-particular. 
- Also the test1.csv which has no keywords and causes an unhandled error in Data_Importer_Pre_Processing_Example.ipynb
- Also tests fail locally if example notebooks have been run

### CODE REVIEW

**Comment by PipGrylls** [comment]
This will close #18

**Comment by PipGrylls** [comment]
This PR will also remove MPL and tidy requirements a bit. Lumping it in as it was basically done anyway by this refactor.

File: README.md [hunk 1]
```diff
@@ -2,5 +2,70 @@
...
 [![codecov](https://codecov.io/gh/paramm-team/data_processing/graph/badge.svg?token=7Xmov38bCi)](https://codecov.io/gh/paramm-team/data_processing)
+[![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
 
-**WARNING:** this package is still under development, and still at a very early stage.
\ No newline at end of file
+**WARNING:** this package is still under development.
+
+This package is provides parameter optimization for PyBaMM (Python Battery Mathematical Modelling) using different optimization techniques. Examples on how to run this package can be found in the [examples folder](./examples)
+
+## 🚀 Installing pybamm-param
```
Comment (brosaplanella) [inline README.md]:
```suggestion
## 🚀 Installing python-battery-data-processing
```

Comment (brosaplanella) [reply README.md]: I like the name that Pip suggested: `python-battery-data-processing` which abbreviates to `pbdp` (which is available in PyPI I believe).

@AlexandruPascu, what do you think?

File: README.md [hunk 2]
```diff
@@ -2,5 +2,70 @@
 
 [![Tests](https://github.com/paramm-team/data_processing/actions/workflows/test_on_push.yml/badge.svg)](https://github.com/paramm-team/data_processing/actions/workflows/test_on_push.yml/badge.svg)
 [![codecov](https://codecov.io/gh/paramm-team/data_processing/graph/badge.svg?token=7Xmov38bCi)](https://codecov.io/gh/paramm-team/data_processing)
+[![Code style: black](https://img.shields.io/badge/code%20style-black-000000.svg)](https://github.com/psf/black)
 
-**WARNING:** this package is still under development, and still at a very early stage.
\ No newline at end of file
+**WARNING:** this package is still under development.
+
+This package is provides parameter optimization for PyBaMM (Python Battery Mathematical Modelling) using different optimization techniques. Examples on how to run this package can be found in the [examples folder](./examples)
```
Comment (brosaplanella) [inline README.md]:
```suggestion
This package provides tools to process and handle battery cycler experimental data. Examples on how to run this package can be found in the [examples folder](./examples)
```

Comment (brosaplanella) [reply README.md]: These changes also need to be applied in the other PR.

**Review by brosaplanella — APPROVED** [review]
Looks good to me, just a couple small comments in the README.

@AlexandruPascu feel free to take a look and leave your comments if you want. The main point of this PR is to switch from `os` to `pathlib` which is more pythonic, but functionality is the same.

File: README.md (hunk 1 above)
Comment (AlexandruPascu) [reply README.md]: The name is nice, I agree and seems like a bonus if it's also available in PyPI!

File: README.md (hunk 2 above)
Comment (AlexandruPascu) [reply README.md]: Nice rewording.

**Review by brosaplanella — APPROVED** [review]

**Pull request merged**